### PR TITLE
fix: various simbrief route import failures

### LINKED ISSUE #2923: [BUG] SimBrief flight plan not correctly entered: AWY/WPT MISMATCH
**Mod Version**

2021-01-07T01:08:05+00:00

**Describe the bug**

The SimBrief flight plan between FAOR to FYWH with route
VASU3A VASUR UQ24 ETOSA G653 GBV UN182 AXODO AXOD1B

Is not correctly entered

**To Reproduce**

1. On SimBrief, define a flight plan between FAOR to FYWH with route
VASU3A VASUR UQ24 ETOSA G653 GBV UN182 AXODO AXOD1B
2. On MCDU when init request, WY/WPT MISMATCH appears and the FP has no intermediate waypoints
3.

**Expected behavior**

FP entered shall be similar to SimBrief definition

**Actual behavior**

Intermediate waypoints of SimBrief Flight Plan are not entered

**References**

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?
Not tested
I used also Navigraph data cycle 2014 rev2

Discord username (if different from GitHub):

### LINKED ISSUE #3385: [BUG] SimBrief Flight Plan: coordinates not works in A320
**Mod Version**

2021-02-06T20:19:15+00:00

**Describe the bug**

Wapypoints defined in coordinates (lat,lon) don't work

**To Reproduce**

1. In SimBrief, use this route   GCXO/30 TFS2D TFS DCT 280030N0170518W DCT 280230N0170842W DCT GCGM/27
2.
3.

**Expected behavior**

The two waypoints 280030N0170518W DCT 280230N0170842W should be entered in MCDU

**Actual behavior**

the two waypoint are not entered

**References**

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

### LINKED ISSUE #3403: [BUG] Loading Simbrief routes and STARs with actual AIRAC circle are "Not in Database"
**Mod Version**
{
  "built": "2021-02-07T05:52:09+00:00",
  "ref": "master",
  "sha": "0f2481ad9161e49a97ed3a379a5f8c3293e42461",
  "actor": "tyler58546",
  "event_name": "manual"
}

**Describe the bug**

Created in SB a flight with the actual AIRAC circle 2101 from EINN->ESSD

**AGIN3A AGINI DCT ERNAN DCT MAC UN537 GOW L983 SAB UM89 RIVOT UM82 NIVUN P609 KSD DCT XOTBU XOTB1F**

When loading the flight in the MCDU INIT1 Page (after _Init Data Request"_ in MCDU ATSU), the message AWY WYP Not in Database was shown. Add in the FP the Rwy's and SID/STAR, checking the FP, the route was _shorten_ to **AGIN3A AGINI DCT ERNAN DCT XOTBU XOTB1F _(but without the WP of the STAR)_ and the ILS Z 32 WP**. Add the WYP MAC manual as _Next Waypoint_ for the ERNAN successfuly and try to add from MAC the AW _UN357_ with no success. So I added as Next Waypoint all the "Endpoints" of the missing Airways (GOW, SAB, RIVOT, NIVUN, KSD). On the Route Display it looks fine, so I start the flight.
During the flight I found, that the "Distance" at the FP-Display of the FMGC was totally different from the ND (see screenshots) and that the total distance to the Airport was "frozen" - no decrease. (see screenshots).
The difference was logical for me for the NIVUN P609 KSD Airway, because I did not entered in the beginning all the waypoints on the P609 (what I did during the flight, after I recognized it on the LiveMap of STKP (SimToolKitPro) and NAVIGRAPH CHARTS) But the screeshot shows the situation between the WP RIVOT UM82 NIVUN which has **NO** additional WPs between RIVOT and NIVUN.
After I passed the XOTBU WP - which is the beginning of the XOTB1F STAR, the plane goes direct to ESSD. I changed this with DIR-TO to the next known WP on the ILS Approach and was luckely able to land the Plane correct, because this behavior is really anoying, especially if you fly Online with IVAO and Real ATC.

Also in the flight from LPAZ-EINN before the mentioned one, with the following SB Route

XUVA1N XUVAP DCT BAVAS DCT **42N022W** DCT **45N020W** DCT KOGAD DCT ATSUR DCT BUNON BUNO2E

both coordinated waypoints are ignored and also reported as "Not in Database". I don't know which MOD Version I used for this flight, I alwas load the latest Dev-Version just before starting the MSFS. The flight starts: 2021-02-06 14:35:48 (if this hepls)

**To Reproduce**

1. USE in SB the actual AIRAC 2101
2. Load via Navigraph Installer for MSFS the actual AIRAC 2101 into the MSFS Community Folder
3. Start Cold/Dark at EINN
4. Load via MCDU the actual SB Flight (OFP)
3. Watch the show ;-)

**Expected behavior**

That the actual AIRAC circle is known and **USED**

**Actual behavior**

see above

**References**

See the difference with the distance

Microsoft Flight Simulator Screenshot 2021 02 07 - 16 45 48 23
Microsoft Flight Simulator Screenshot 2021 02 07 - 16 45 59 74

**Additional context**

Discussed with @pareil6 in Issue #3385

Hope all is understandable, sorry I'm not a native english speaker 

Was this working before/when did the issue start occurring?
Had sometimes Problems with the STARs not flown correct and was wondering that in MCDU the loaded AIRAC was not shown up See #3233 
Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

### LINKED ISSUE #3406: [BUG] SimBrief AWY/WPT MISMATCH failures on valid route segments, correctly loads on second attempt
**Mod Version**
master (2e8f2470)

**Describe the bug**
When importing a route from SimBrief, the import will fail randomly on a valid airway/waypoint intersection with a AWY/WPT MISMATCH warning. Forcing the route to reload will generally fix it - implying that airway data is failing to load correctly on the first attempt. There's no obvious reason for what does it - have seen it reported several times today, with nothing in common between the routes that caused it. An example route (valid on AIRAC 2014/2101):
TTPP DCT POS UG449 ANADA L452 HARBG Y421 OCTAL DCT BAGGS Y290 LEV GILCO5 KIAH

**To Reproduce**
1. Create a SimBrief route from TTPP to KIAH with the above routing
2. Load in, turn on power, and perform an INIT REQUEST from the INIT A page

**Expected behavior**
Route correctly populates

**Actual behavior**
INIT A page populates, but an AWY/WPT MISMATCH shows, with the route failing to load past POS in the F-PLN page, warning in the MCDU console:

On clearing the route and reloading from SimBrief (typing TTPP/TTPP in FROM/TO to allow another INIT REQUEST, then making a request), the route fills correctly, console logs:

Note that very occasionally, it takes multiple reload attempts to make it work - may potentially be related to the same bug that causes a NOT IN DATABASE error when entering a FROM/TO that then works on the next attempt?

**References**
N/A

**Additional context**

Was this working before/when did the issue start occurring? Seems to have occurred sporadically in the past, but has been reported more frequently lately

Is this a problem in the vanilla unmodded game? N/A

Discord username (if different from GitHub): pareil6#0990

### LINKED ISSUE #3444: [BUG] Need to enter and validate altitude constrants 2 times
Mod Version
"built": "2021-02-09T07:03:20+00:00"
"ref": "fbw"

**Describe the bug**
When modifying altitude constraints in flight plan before take off, I have to enter and validate the constraint 2 times before it was taken into account.

I do not need to enter and validate 2 times with a previous mod version:
"built": "2021-02-04T06:24:14+00:00"
"ref": "fbw"

**To Reproduce**
1. Click to edit the altitude constraint on a waypoint
2. Enter a constraint (ex: 100 for FL100 or 10000) and INSERT -> Nothing is changed
3. Enter this constraint again (ex: 100 for FL100 or 10000) and INSERT -> Constraint is taken into account

**Expected behavior**
I should be able to enter and validate the constraint the first time.

**Actual behavior**
The constraint is taken into account the second time.

### PR DESCRIPTION
**Pull request by pareil6**
Fixes #2923, fixes #3385, fixes #3403, fixes #3444.

## Summary of Changes
Fixes various issues with importing SimBrief routes:

- Support for coordinates, parsing the format used by SimBrief into the format required by MSFS (#3385)
- Correctly handle SIDs with a single fix, instead of trying to use the SID as an airway (#3403)
- Modify airway loading when importing from SimBrief to search by airway name instead of loading all airways passing through the previous waypoint (to match manual import logic)
- Update FlightPlanManager not to rely on SimVar version number, fixing cases where updates would fail to occur causing waypoint additions to go missing briefly - fixes vast majority of AWY/WPT MISMATCH errors currently occuring, and also fixes altitude constraints not appearing on first attempt to insert them (#2923, #3444)

## Screenshots (if necessary)
N/A (no visual changes)

## References
N/A

## Additional context

Discord username (if different from GitHub): pareil6#0990

## Testing instructions
Check I've not broken any normal SimBrief behaviour on a normal route. Coordinate logic is the only properly new thing here - try some routes with coordinates in them (in all 4 quadrants if possible, ie. xxNxxxE/xxNxxxW/xxSxxxE/xxSxxxW). If you've got any routes that have failed to load via SimBrief integration, try them again with this PR. Also check I haven't broken anything flight plan management related - eg. make sure the NDs stay synced with the MCDU flight plan, that approaches/directs/etc. don't mysteriously fail to load/cancel (or behave any worse than they did before), etc.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by pareil6** [comment]
I've noticed this happening on a lot more routes recently, seemingly with a couple of waypoints being inserted in the wrong order (example was DCS UN864 SUBUK DCT KEPAD - flight plan in the MCDU would contain KEPAD SUBUK then would fail, so it looks like there's an async call happening somewhere that means the faster DCT insert can complete before the slower airway insert?). Wonder if it's related to #3262 after all - TMPY flight plans also seem to take a lot longer to update...

**Comment by pareil6** [comment]
On the remaining AWY/WPT MISMATCH errors - we seem to occasionally start trying to load a later route segment before an earlier one has been synced properly into the FPM (most often an airway segment following a DCT segment). Seems we can fudge around this by undoing the updateFlightPlan() optimisation from #3262 when calling it from addWaypoint() - looks like there's an underlying race condition somewhere anyway, and that performance fix exposed it far more frequently. Will push a fix later.

(cc @XSJoJo - not sure if you've seen similar issues and/or already have a bugfix that's likely to fix this)

**Comment by XSJoJo** [comment]
> On the remaining AWY/WPT MISMATCH errors - we seem to occasionally start trying to load a later route segment before an earlier one has been synced properly into the FPM (most often an airway segment following a DCT segment). Seems we can fudge around this by undoing the updateFlightPlan() optimisation from #3262 when calling it from addWaypoint() - looks like there's an underlying race condition somewhere anyway, and that performance fix exposed it far more frequently. Will push a fix later.
> 
> (cc @XSJoJo - not sure if you've seen similar issues and/or already have a bugfix that's likely to fix this)

Oh interesting - are you saying that the GetSimVar call returns the old value after calling _incrementFlightPlanVersion()? That seems..unexpected! If that's the case we probably need to use forceUpdate for all the places where we increment the version.

**Comment by pareil6** [comment]
> Oh interesting - are you saying that the GetSimVar call returns the old value after calling _incrementFlightPlanVersion()? That seems..unexpected! If that's the case we probably need to use forceUpdate for all the places where we increment the version.

I'm not actually sure - when I checked the return value it always seemed correct, but occasionally inserting a waypoint would still end up with an old FP version number. I'm wondering if there's an async call to the FPM somewhere that means the version number change happens out of sequence? I'll have a closer look later, but that's my best guess at the moment...

**Comment by pareil6** [comment]
Actually, I've just noticed adding altitude constraints now has some strange behaviour as well - it takes two attempts to add an altitude constraint now. Will debug to see if it's the same issue.

**Comment by pareil6** [comment]
> Oh interesting - are you saying that the GetSimVar call returns the old value after calling _incrementFlightPlanVersion()? That seems..unexpected! If that's the case we probably need to use forceUpdate for all the places where we increment the version.

@XSJoJo - to answer this question, unfortunately, yes...

```
    _incrementFlightPlanVersion() {
        // Get most up to date version in case updateFlightPlan hasn't been called yet before the last increment.
        const currentVersion = SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number');
        SimVar.SetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number', currentVersion + 1);
        console.log(currentVersion + " " + SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number'));
    }
```

...and...

```
    updateFlightPlan(callback = () => { }, log = false) {
        const newVersion = SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number');
        console.log(newVersion);
        ....
```

...gives me this:



ie. the increment doesn't register on the next couple of GetSimVarValue calls at least.

I think we _generally_ get away with it because the vast majority of flight plan changes are either manually adding/removing a waypoint, or a waypoint being passed on the flight plan. Both of these make changes less frequently than the timed updateFlightPlan() calls occur - ie. when you type an airway in manually, by the time you've typed the next one in, the function has been called again anyway, detecting the version change and regenerating. The only place we consistently see a problem is with SimBrief imports, where we programmatically add waypoints, meaning each waypoint gets added potentially before the SimVar has properly updated, so we end up with some intermediate versions getting skipped, leaving us with the wrong last waypoint everywhere we rely on it.

Is the flight plan version stored as a SimVar so it's visible from the ND? I'm currently testing tracking the flight plan version as a variable within the FPM for local updates that synchronises to the SimVar in _incrementFlightPlanVersion() calls, on the assumption that the ND can safely lag the MCDU view - just wondering if that potentially breaks anything else?

**Comment by XSJoJo** [comment]
> @XSJoJo - to answer this question, unfortunately, yes...

Ah great, thanks for investigating this. Looks like SimVar is a multi-core data sharing mechanism where not even the own core sees the latest state, and the entire system isn't multi-threaded anyways 🤔

We should probably force the reload for all calls where we update the flight plan then.

**Comment by pareil6** [comment]
I've tweaked how we store the flight plan version so we store it as an FPM class member instead, and export the SimVar when it changes - which seems to have fixed the loading issues I was seeing while still throttling any unnecessary updates.

File: A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/LogicElements/A32NX_FlightPlanManager.js [hunk 1]
```diff
@@ -395,20 +396,18 @@ class FlightPlanManager {
...
-        const currentVersion = SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number');
-        SimVar.SetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number', currentVersion + 1);
+        this._newFlightPlanVersion++;
+        SimVar.SetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number', this._newFlightPlanVersion);
     }
 
     updateFlightPlan(callback = () => { }, log = false) {
-        const newVersion = SimVar.GetSimVarValue("L:A32NX_FLIGHT_PLAN_VERSION", 'number');
-        if (newVersion === this._currentFlightPlanVersion) {
+        if (this._newFlightPlanVersion === this._currentFlightPlanVersion) {
```
Comment (XSJoJo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/LogicElements/A32NX_FlightPlanManager.js:404]: you also need to check the sim var value here, as this is the way FlightPlanManager from MCDU and ND communicate with each other - there are 3 instances: MCDU has one, Caption ND has one, and FO ND has one.

Comment (pareil6) [reply A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/LogicElements/A32NX_FlightPlanManager.js:494]: Fixed, thanks!

**Comment by koenvanderlinden** [comment]
I don't know if this is the right place to comment on some related simbrief routing problems? I've been working on some issues I noticed with simbrief routing. I used some code of this PR as a base.

When testing/debugging I noticed an issue adding a waypoint using insertWaypointsAlongAirway with a 'via' object. It's not working when the routeIdent has multiple waypoints. For example when adding simbrief route (Orig: LEBB, Dest: EHAM), routing: BISK3A BISKA P87 BELEN UP87 MOKOR UN873 FERDI UY18 DENUT DENU1A. It fails adding FERDI with via 'UN873'. When you do a lookup by hand on FERDI it returns a choice between two waypoints.

I implemented a fallback when insertWaypointsAlongAirway fails. I do a add waypoint without a via. The simbrief route is then inserted correctly, except that FERDI has no via information. BTW, same goes for DENUT via UY18.

Using the interface to insert FERDI via UN873 also does not work. But I have not checkt if this is custom FBW code or original code.

Solution would be that insertWaypointsAlongAirway should take into account that the Indent could have multiple hits.
Do you want to fix this? Or can I contribute in some way?

**Comment by pareil6** [comment]
@koenvanderlinden - it sounds like your solution will effectively mean if an airway is invalid, we'll fall back to adding it as a DCT? It's debatable whether this is actually better - if the airway can't be found, I think I prefer the route failing to insert to force the user to go back to figure out why it's not valid, rather than silently changing it (which will potentially be an issue if you're flying on VATSIM/IVAO and you've filed the route via the airway).

Incidentally, the route example you've given is actually invalid on the latest AIRAC - UN873 doesn't exist any more (as of AIRAC 2101), and has been consolidated with the low-level airway which is identical (N873). This is a surprisingly common problem - I think there was an effort to remove unnecesary upper/lower airway splits in western Europe somewhere between cycle 1903 and 2101, so that route should work if you replace UN873 with N873.

(I only know that because I spent an afternoon debugging exactly the same thing on a route containing UL15, which has also been replaced with making L15 valid in upper airspace too...)

**Review by XSJoJo — APPROVED** [review]
FPM changes LGTM.

**Comment by koenvanderlinden** [comment]
@pareil6 thanks for explaining, I’m a bit new to these terms. So the problem is mismatch in AIRAC you use in Simbrief and the one is used in MSFS. I don’t have access to other than the free AIRAC on simbrief.
Indeed, fallback mechanisme is not a good solution.

**Review by beheh — APPROVED** [review]

**Comment by BritishAvgeek** [comment]
Category: QA Tester
Name: BritishAvgeek#1436
Date of testing: 12/02/2021
Version of the sim: 1.12.13.0
PR Tested: #3407 
Tier of testing: 2
Changes to observe: fix import simbrief failures, add co-ords


Testing technique: Spawned C&D at LPPT - prepped for oceanic flight with one co-ord
Steps you've made to test the Pull Request:
1.  Created route from LPPT to LPPD with one oceanic co-ordinate waypoint along the way to test additional feature
2.  Spawned C&D, and began import of simbrief routing and checked for any issues throughout
3.  Checked co-ordinate was imported also and the sim recognised it in the Plan on ND and MCDU
4. Added SID/STAR to check any issues

Flight notes (When done a full flight): N/A

Media: Insert screenshots, recordings, or anything relevant to the review here.
image
image
image
image
image
image
image
image


Issues (If Any): Cost Index of 25 didn't import like it did a while back but everything else was perfect and worked very well - the oceanic waypoint went in with zero issue too!
Steps to achieve the issue

Overall Rating: Good
Conclusions of the testing: Small issue of the cost index not going in (I'm sure it did in the past but may be wrong!!) tested it with the oceanic waypoint and it was absolutely fine and added that co-ordinate in perfectly!

**Comment by BritishAvgeek** [comment]
Category: QA Tester
Name: BritishAvgeek#1436
Date of testing: 12/02/2021
Version of the sim: 1.12.13.0
PR Tested: #3407
Tier of testing: 2
Changes to observe: fix import simbrief failures, add co-ords

Testing technique: C&D at EGLL - set up for flight onwards to KJFK with a NAT Track to check new co-ord import input too
Steps you've made to test the Pull Request:
1.  Generated a complex oceanic route using simbrief and NAT A to see how the co-ords would handle with the route showing as "NATA" on simbrief
2.  Imported it and it didn't work properly - skipped it all
3.  I then restarted, and re-generated the simbrief route and physically typed the co-ordinated into the flight plan on simbrief instead of it using "NAT A" or using the NAT A co-ord format within the flight plan
4. Imported the flight plan into MCDU using the "workaround" shown - imported perfectly!

Flight notes (When done a full flight): N/A

Media: Insert screenshots, recordings, or anything relevant to the review here.
image
image
image


Issues (If Any): NAT Tracks input via simbrief don't currently get recognised so it skips a lot of it as shown in the screenshots
Steps to achieve the issue - produce a route from EGLL to KJFK with a NAT track which generates a couple of co-ordinates in a different format on simbrief - it triggered "AWY/WPT MISMATCH" in the scratchpad during Init A import for the route

I then went BACK into simbrief and manually input each co-ordinate within the NAT track instead to have them all individually input and the input worked fine again - it imported each one perfectly!
image


Overall Rating: To be Improved if at all possible!
Conclusions of the testing: Co-ordinates work unless linked into a NAT Track due to the way Simbrief actually produces them in a route - not sure if it is even possible in fairness but I found the workaround above where you type the actual co-ords out in the planning section on simbrief

**Comment by pareil6** [comment]
>Co-ordinates work unless linked into a NAT Track due to the way Simbrief actually produces them in a route - not sure if it is even possible in fairness but I found the workaround above where you type the actual co-ords out in the planning section on simbrief

So I was going to do this, but it's a bit awkward because it means unpicking the existing add-along-airway code specially for this case - and to be completely honest, if I'm doing it for that case, I'd rather bite the bullet and redo it for all cases, ie. instead of searching along the navlog to find each fix -> airway -> fix segment then using the nav data to add the fixes in between, just add all the navlog fixes directly, and *warn* the user if the airway we expect to be on isn't the one we're actually on. This has the side effect of hiding all the cases we currently fail on where the upper airways have merged with the lower airways while still producing a correct route, and means I can remove the airway search code from the SimBrief import entirely.

This is my fault for rolling what's really a feature into a bugfix - but I'd prefer not blocking the FPM versioning fix to do that here, just because the AWY/WPT MISMATCH thing keeps coming up in Discord. I'll do a separate PR to clean that up and push it separately, unless anyone would prefer I didn't.

**Comment by BritishAvgeek** [comment]
I think it's perfectly fine as it is to be honest - the airway feature would be good ability to have added on at least for some oceanic waypoint and "random routing"... can put it out that if using NAT Tracks then the co-ords need to be added in full into the simbrief flight plan for the specifics of the simbrief integration feature added here? basically took me about 30 seconds to add it all in extra myself

**Comment by PierreLabs** [comment]
**Category**: QA Test
**Discord Name**: Zephyr#0959
**Date of testing**: 14/02/2021
**Version of the sim**: 1.12.13.0
**PR Tested**: #3407
**Changes to observe**: fix for simbrief import (no WPT/MISMATCH when importing)

Testing technique: Cold & dark at LFPO. importing a flight to EGNT.
Steps for the test:
0. Spawn at LFPO gate P10
1. Battery switches ON, EXT PWR ON.
2. ATSU -> AOC MENU
3. INIT/PRES -> INIT DATA REQ
4. PERF/W&B -> REFUEL LOAD -> page2 -> PAYLOAD LOAD
5. INIT -> INIT REQUEST -> input missing COST INDEX
6. insert SID and STAR
7. check FP for any error/inconsistency
8. back to main menu and to step 0 to test a second time

Media:
LFPOEGNTfp
LFPOEGNT1
LFPOEGNT
LFPOEGNT3


Issues: Cost index didn't import like it did before as spotted by @brit

Overall Rating: Good

Conclusions: I tried importing this FP with the cfbw build and had it fail 4 times in a row (reinit the request after entering random ICAOs in the FROM/TO).
With this PR, it worked the first time twice (back to main menu and did the test again).

**Comment by BritishAvgeek** [comment]
Category: QA Tester
Name: BritishAvgeek#1436
Date of testing: 14/02/2021
Version of the sim: 1.12.13.0
PR Tested: #3407 
Tier of testing: 2
Changes to observe: simbrief import failures fix - This is my **third test for this PR** so I'm confident it works well

Testing technique: Full Flight on VATSIM from cold and dark
Steps you've made to test the Pull Request:
1.  Spawned EDDM cold and dark, turned BAT1 on, then APU Master/APU Start, APU Bleed and BAT2 this time rather than using Ext Power
2. Imported the route as shown below via simbrief, worked perfectly fine, no issues or missed wpts etc
3. Flew the route on VATSIM and did a direct to within the STAR to see if that would break - no issues
4. Performed a go-around due to simply horrendous weather on short finals - AP bugged out as usual but the route remained visible and had no issues. MCDU went into G/A mode and back into APPR mode when I selected it to on the downwind leg
5. Landed

Flight notes (When done a full flight):
Route: EDDM 08R/GIVM6E GIVMI Y101 INBED DCT REKDI DCT LOHRE T880 ADMUM/N0426F240 T880 ASDAK DCT TERLA L608 DENUT L610 HEMEL T420 WELIN (HEMEL1B) ILS15/EGBB

Media: Livestreamed here https://youtu.be/9ooyKRv0rFY

Issues (If Any): None

Overall Rating: Good
Conclusions of the testing: No issues at all - Cost Index didn't import but that hasn't done  so for a few weeks, conversation on QA chat the other day makes me thing something minor inside the simbrief api may have changed but everything worked really well here - no issues with the import.

Over three tests I've now conducted for this PR, I've imported a single co-ordinate (new feat within the importation), used an expanded NAT Track (typed each co-ord separately out into the plan on simbrief which imported wonderfully for an oceanic LHR-JFK) and now done a full flight from EDDM to EGBB "pan-european" to check all types of flight plan likely to be entered into this and had no issues with it.

**Comment by pareil6** [comment]
Thanks @BritishAvgeek - I've just checked again locally and the CI doesn't autofill on current master either, so it's definitely not a regression caused by this PR.

**Review by kberg93 — APPROVED** [review]
Category: QA Tester
Name: Kberg#0666
Date of testing: Current date 02/14/2021
Version of the sim: 1.12.13.0
PR Tested: #3407 
Tier of testing: 2
Changes to observe: Simbrief route import failures.

Testing technique: Importing flightplan from Simbrief and flying it.

Steps you've made to test the Pull Request: 
1. C&D at5 ENVA importing the flightplan from Simbrief.
2. No issues with flightplan nor fuel and pax.
3. Take off and flew to Svalbard without any issue.
4. Landed and shut down the plaane.

Flight notes (When done a full flight): Route= ENVA NIGE1X NIGEX Z103 BNN M609 STO Z202 SJA M609 TRO P600 LOMVI/N0451F340 P853 UVSAS/N0450F350 DCT INPAR DCT ENSB

Media: Insert screenshots, recordings, or anything relevant to the review here.

Issues (If Any): None.

Overall Rating: Good.
Conclusions of the testing: None issues that i could find or point fingers at, it went flawless after i found out why i had CTD wich was a sceneryissue not anything with this PR.

**Pull request merged**